cache/refs.go

```diff
@@ -283,6 +284,30 @@ type immutableRef struct {
 	descHandlers    DescHandlers
 }
 
+type RefList []ImmutableRef
+
+func (l RefList) Release(ctx context.Context) (rerr error) {
+	for i, r := range l {
+		if r == nil {
+			continue
+		}
+		if err := r.Release(ctx); err != nil {
+			rerr = multierror.Append(rerr, err).ErrorOrNil()
+		} else {
+			l[i] = nil
+		}
+	}
+	return rerr
+}
+
+func (sr *immutableRef) LayerChain() RefList {
+	var l RefList
+	for _, rec := range sr.layerChain() {
+		l = append(l, rec.ref(false, sr.descHandlers))
```

Comment:
My thinking was that `Parents()` will continue to return slice with one element for non-merge refs. For merge refs it would be only merged parents. Returning all seem unnecessary for API and also too heavy. Reference counting refs should not become slower if there are more layerss.